internal/rbd/rbd_util.go

```diff
@@ -58,125 +58,129 @@ const (
 	thickProvisionMetaKey = ".rbd.csi.ceph.com/thick-provisioned"
 )
 
+// rbdImage contains common attributes and methods for the rbdVolume and
+// rbdSnapshot types.
+type rbdImage struct {
+	// RbdImageName is the name of the RBD image backing this rbdVolume.
+	// This does not have a JSON tag as it is not stashed in JSON encoded
+	// config maps in v1.0.0
+	RbdImageName string
+	// ImageID contains the image id of the image
+	ImageID string
+	// VolID is the volume ID that is exchanged with CSI drivers,
+	// identifying this rbd image
+	VolID string `json:"volID"`
+
+	Monitors string
+	// JournalPool is the ceph pool in which the CSI snapshot Journal is
+	// stored
+	JournalPool string
+	// Pool is where the image snapshot journal and snapshot is stored, and
+	// could be the same as `JournalPool` (retained as Pool instead of
+	// renaming to ImagePool or such, as this is referenced in the code
+	// extensively)
+	Pool           string
+	RadosNamespace string
+	ClusterID      string `json:"clusterId"`
+	// RequestName is the CSI generated volume name for the rbdVolume.
+	// This does not have a JSON tag as it is not stashed in JSON encoded
+	// config maps in v1.0.0
+	RequestName string
+	NamePrefix  string
+
+	// encryption provides access to optional VolumeEncryption functions
+	encryption *util.VolumeEncryption
+	// Owner is the creator (tenant, Kubernetes Namespace) of the volume
+	Owner string
+
+	CreatedAt *timestamp.Timestamp
+
+	// conn is a connection to the Ceph cluster obtained from a ConnPool
+	conn *util.ClusterConnection
+	// an opened IOContext, call .openIoctx() before using
+	ioctx *rados.IOContext
+}
```

Comment:
@nixpanic Isnt it good to have a field inside this struct which says the image refers to a snapshot or not ? so that most of the decision can be based on that field. wdyt?

Reply:
These structs should get extended when there is an opportunity to make the code simpler. I do not immediately see where the code gets simplified, or reduce calls to a "is this image based on a snapshot?" function.

As such a field would not be used by this PR, or the snapshot code is cleaned up, I prefer not to add unused fields now.

Reply:
Yeah, we could consider it later. .Thanks!